File: mystic-tales-podcast-frontend/mystic-tales-podcast-platform-web/src/core/types/account.ts
// @ts-nocheck

import type { CurrentAudioFromApi, CurrentAudioUI } from "./audio";

export type AccountFromAPI = {
  Id: number;
  Email: string;
  Role: AccountRole;
  FullName: string;
  Dob: string; //Format "YYYY-MM-DD"
  Gender: string;
  Address: string;
  Phone: string;
  Balance: number;
  MainImageFileKey: string;
  IsVerified: boolean;
  GoogleId: string | null;
  PodcastListenSlot: number;
  ViolationPoint: number;
  ViolationLevel: number;
  LastViolationPointChanged: string | null; // Format ISO String
  LastViolationLevelChanged: string | null; // Format ISO String
  LastPodcastListenSlotChanged: string | null;
  DeactivatedAt: string | null; // Format ISO String
  CreatedAt: string; // Format ISO String
  UpdatedAt: string; // Format ISO String
  IsPodcaster: boolean; //Để check xem User có phải podcaster không, hay là User thường
};

export type AccountMeFromApi = {
  Id: number;
  Email: string;
  FullName: string;
  Dob: string;
  Gender: string;
  Address: string;
  Phone: string;
  Balance: number;
  MainImageFileKey: string;
  PodcastListenSlot: number;
  DeactivatedAt: string;
  IsPodcaster: boolean;
};

export type AccountMeUI = {
  Id: number;
  Email: string;
  FullName: string;
  Dob: string;
  Gender: string;
  Address: string;
  Phone: string;
  Balance: number;
  ImageUrl: string;
  PodcastListenSlot: number;
  DeactivatedAt: string;
  IsPodcaster: boolean;
};

// Customer: {Id: 1, Name: "string"}
export type AccountRole = {
  Id: number;
  Name: string;
};

// Lấy /me từ API
// export type AccountMeFromApi = {
//   Account: AccountFromAPI;
//   CurrentAudio: CurrentAudioFromApi;
// };

File: mystic-tales-podcast-backend/user-service/UserService.DataAccess/Entities/SqlServer/Account.cs
using System;
using System.Collections.Generic;

namespace UserService.DataAccess.Entities.SqlServer;

public partial class Account
{
    public int Id { get; set; }

    public string Email { get; set; } = null!;

    public string Password { get; set; } = null!;

    public int RoleId { get; set; }

    public string FullName { get; set; } = null!;

    public DateOnly? Dob { get; set; }

    public string? Gender { get; set; }

    public string? Address { get; set; }

    public string? Phone { get; set; }

    public decimal Balance { get; set; }

    public string? MainImageFileKey { get; set; }

    public bool IsVerified { get; set; }

    public string? GoogleId { get; set; }

    public string? VerifyCode { get; set; }

    public int? PodcastListenSlot { get; set; }

    public int ViolationPoint { get; set; }

    public int ViolationLevel { get; set; }

    public DateTime? LastViolationPointChanged { get; set; }

    public DateTime? LastViolationLevelChanged { get; set; }

    public DateTime? LastPodcastListenSlotChanged { get; set; }

    public DateTime? DeactivatedAt { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime UpdatedAt { get; set; }

    public virtual ICollection<AccountFavoritedPodcastChannel> AccountFavoritedPodcastChannels { get; set; } = new List<AccountFavoritedPodcastChannel>();

    public virtual ICollection<AccountFollowedPodcastShow> AccountFollowedPodcastShows { get; set; } = new List<AccountFollowedPodcastShow>();

    public virtual ICollection<AccountFollowedPodcaster> AccountFollowedPodcasterAccounts { get; set; } = new List<AccountFollowedPodcaster>();

    public virtual ICollection<AccountFollowedPodcaster> AccountFollowedPodcasterPodcasters { get; set; } = new List<AccountFollowedPodcaster>();

    public virtual ICollection<AccountNotification> AccountNotifications { get; set; } = new List<AccountNotification>();

    public virtual ICollection<AccountSavedPodcastEpisode> AccountSavedPodcastEpisodes { get; set; } = new List<AccountSavedPodcastEpisode>();

    public virtual ICollection<PasswordResetToken> PasswordResetTokens { get; set; } = new List<PasswordResetToken>();

    public virtual ICollection<PodcastBuddyReview> PodcastBuddyReviewAccounts { get; set; } = new List<PodcastBuddyReview>();

    public virtual ICollection<PodcastBuddyReview> PodcastBuddyReviewPodcastBuddies { get; set; } = new List<PodcastBuddyReview>();

    public virtual PodcasterProfile? PodcasterProfile { get; set; }

    public virtual Role Role { get; set; } = null!;
}
